packages/frontend/component/src/ui/editable/inline-edit.tsx

```diff
@@ -0,0 +1,226 @@
+import clsx from 'clsx';
+import {
+  type CSSProperties,
+  type ForwardedRef,
+  type HTMLAttributes,
+  type PropsWithChildren,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+
+import Input from '../input';
+import * as styles from './inline-edit.css';
+
+export interface InlineEditHandle {
+  triggerEdit: () => void;
+}
+
+export interface InlineEditProps
+  extends Omit<HTMLAttributes<HTMLSpanElement>, 'onChange' | 'onInput'> {
+  /**
+   * Content to be displayed
+   */
+  value?: string;
+  /**
+   * Whether the content is editable
+   */
+  editable?: boolean;
+
+  onInput?: (v: string) => void;
+  onChange?: (v: string) => void;
+
+  /**
+   * Trigger edit by `click` or `doubleClick`
+   * @default `'doubleClick'`
+   */
+  trigger?: 'click' | 'doubleClick';
+
+  /**
+   * whether to auto select all text when trigger edit
+   */
+  autoSelect?: boolean;
+
+  /**
+   * Placeholder when value is empty
+   */
+  placeholder?: string;
+  /**
+   * Custom placeholder `className`
+   */
+  placeholderClassName?: string;
+  /**
+   * Custom placeholder `style`
+   */
+  placeholderStyle?: CSSProperties;
+
+  handleRef?: ForwardedRef<InlineEditHandle>;
+
+  /**
+   * Customize attrs for the input
+   */
+  inputAttrs?: Omit<HTMLAttributes<HTMLInputElement>, 'onChange' | 'onBlur'>;
+}
+
+export const InlineEdit = ({
+  value,
+  editable,
+  className,
+  style,
+  trigger = 'doubleClick',
+  autoSelect,
+
+  onInput,
+  onChange,
+
+  placeholder,
+  placeholderClassName,
+  placeholderStyle,
+
+  handleRef,
```

Comment:
I think `useImperativeHandle` should be used with `forwardRef` instead of passing as a prop.